tidb-performance-tuning-config.md

```diff
@@ -0,0 +1,533 @@
+
+    ```sql
+    SET GLOBAL tidb_max_chunk_size = 4096;
+    ```
+
+### Optimize transaction mode and DML type for different workloads
+
+TiDB provides different transaction modes and DML execution types to optimize performance for various workload patterns.
+
+#### Transaction modes
+
+You can set the transaction mode using the [`tidb_txn_mode`](/system-variables.md#tidb_txn_mode) system variable.
+
+- [Pessimistic transaction mode](/pessimistic-transaction.md) (default):
+
+    - Suitable for general workloads with potential write conflicts.
+    - Provides stronger consistency guarantees.
+
+  ```sql
+  SET SESSION tidb_txn_mode = "pessimistic";
+  ```
+
+- [Optimistic transaction mode](/optimistic-transaction.md):
+
+    - Suitable for workloads with minimal write conflicts.
+    - Better performance for multi-statement transactions.
+    - Example: `BEGIN; INSERT...; INSERT...; COMMIT;`.
+
+  ```sql
+  SET SESSION tidb_txn_mode = "optimistic";
+  ```
+
+#### DML types
+
+You can control the execution mode of DML statements using the [`tidb_dml_type`](/system-variables.md#tidb_dml_type-new-in-v800) system variable, which is introduced in v8.0.0.
+
+To use the bulk DML execution mode, set `tidb_dml_type` to `"bulk"`. This mode optimizes bulk data loading without conflicts and reduces memory usage during large write operations. Before using this mode, ensure that:
+
+- Auto-commit is enabled.
+- The [`pessimistic-auto-commit`](/tidb-configuration-file.md#pessimistic-auto-commit-new-in-v600) configuration item is set to `false`.
+
+```sql
+SET SESSION tidb_dml_type = "bulk";
+```
+
+### Optimize `GROUP BY` and `DISTINCT` operations with TiKV pushdown
+
+TiDB pushes down aggregation operations to TiKV to reduce data transfer and processing overhead. The performance improvement varies based on your data characteristics.
+
+#### Usage scenarios
+
+- **Ideal scenarios** (high performance gain):
+    - Columns containing few distinct values (low NDV).
+    - Data containing frequent duplicate values.
+    - Example: status columns, category codes, date parts.
+
+- **Non-ideal scenarios** (potential performance loss):
+    - Columns containing mostly unique values (high NDV).
+    - Unique identifiers or timestamps.
+    - Example: User IDs, transaction IDs.
+
+#### Configuration
+
+Enable pushdown optimizations at the session or global level:
+
+```sql
+-- Enable regular aggregation pushdown
+SET GLOBAL tidb_opt_agg_push_down = ON;
+
+-- Enable distinct aggregation pushdown
+SET GLOBAL tidb_opt_distinct_agg_push_down = ON;
+```
+
+### Mitigate MVCC version accumulation using in-memory engine
+
+Excessive MVCC versions can cause performance bottlenecks, particularly in high read/write areas or due to issues with garbage collection and compaction. You can use the [in-memory engine](/tikv-in-memory-engine.md) introduced in v8.5.0 to mitigate this issue. To enable it, add the following configuration to your TiKV configuration file.
```

Comment:
```suggestion
Excessive MVCC versions can cause performance bottlenecks, particularly in high read/write areas or due to issues with garbage collection and compaction. You can use the in-memory engine introduced in v8.5.0 to mitigate this issue. To enable it, add the following configuration to your TiKV configuration file.
```

The PR containing `tikv-in-memory-engine.md` has not been merged into the master branch.